### PR TITLE
MYST-508 Mysterion logs

### PR DESCRIPTION
**Pull request by Waldz**

### CODE REVIEW

File: src/app/mysterion.js [hunk 1]
```diff
@@ -26,26 +28,40 @@ class Mysterion {
...
       }
-      this.window.show()
     })
     app.on('before-quit', () => {
       this.window.willQuitApp = true
     })
   }
 
   async bootstrap () {
+    logInfo('Checking terms cache')
```
Comment (ignasbernotas) [inline src/app/mysterion.js]: Change to "Bootstrapping application"?

File: src/app/mysterion.js [hunk 2]
```diff
@@ -60,60 +76,71 @@ class Mysterion {
       browserWindow = this.browserWindowFactory()
       this.window = this.windowFactory()
       if (termsAccepted) {
+        logInfo('Opening application window')
```
Comment (ignasbernotas) [inline src/app/mysterion.js]: "Terms accepted, resizing the window to app window size"

File: src/libraries/mysterium-client/launch-daemon/process.js [hunk 3]
```diff
@@ -17,9 +17,13 @@ class Process {
   }
 
   start () {
-    this.tequilapi.healthCheck().catch(() => {
-      console.log('touched the daemon, now it should be up')
-    })
+    this.tequilapi.healthCheck()
+      .then(() => {
+        console.log('Touched the daemon with error, anyway it should be up')
```
Comment (ignasbernotas) [inline src/libraries/mysterium-client/launch-daemon/process.js]: Shouldn't these be switched places? Catch callback should say "with error"

File: test/integration/libraries/mysterium-client/standalone/process.spec.js [hunk 4]
```diff
@@ -36,22 +37,44 @@ xdescribe('Standalone Process', () => {
     process.stop()
   })
 
-  it('spawns in less than 100ms without errors', () => {
-    expect(process.child).to.be.instanceOf(ChildProcess)
-  })
+  describe('startup', () => {
+    it('spawns in less than 100ms without errors', () => {
```
Comment (ignasbernotas) [inline test/integration/libraries/mysterium-client/standalone/process.spec.js]: not sure about these 100ms? Maybe increase it to 500 at least?

File: src/app/mysterion.js [hunk 5]
```diff
@@ -26,26 +28,40 @@ class Mysterion {
...
-        return this.bootstrap()
+      try {
+        logInfo('Application activate')
+        if (!this.window.exists()) {
+          this.bootstrap()
+          return
+        }
+        this.window.show()
+      } catch (e) {
+        logException('Application activate failed', e)
```
[the remark this answers is not in the document]
Comment (Waldz) [reply src/app/mysterion.js]: Changed

File: src/app/mysterion.js (hunk 1 above)
Comment (Waldz) [reply src/app/mysterion.js]: But i want to log terms step here

File: src/app/mysterion.js (hunk 2 above)
Comment (Waldz) [reply src/app/mysterion.js]: Changed

File: src/libraries/mysterium-client/launch-daemon/process.js (hunk 3 above)
Comment (Waldz) [reply src/libraries/mysterium-client/launch-daemon/process.js]: Fixed

File: test/integration/libraries/mysterium-client/standalone/process.spec.js (hunk 4 above)
Comment (Waldz) [reply test/integration/libraries/mysterium-client/standalone/process.spec.js]: Not my scope

File: src/app/mysterion.js [hunk 6]
```diff
@@ -25,26 +27,40 @@ class Mysterion {
...
-        return this.bootstrap()
+      try {
+        logInfo('Show window')
+        if (!this.window.exists()) {
+          this.bootstrap()
+          return
+        }
+        this.window.show()
+      } catch (e) {
+        logException('Failed to show window', e)
```
Comment (donce) [inline src/app/mysterion.js]: This is false - there are more things in `try` block than showing a window, i.e. terms, process initialisation and so on.

File: src/app/mysterion.js [hunk 7]
```diff
@@ -25,26 +27,40 @@ class Mysterion {
...
+        this.bugReporter.main.captureException(e)
```
Comment (donce) [inline src/app/mysterion.js]: I'm not 100% sure, but I believe that failed hook will terminate application if this hook throws an exception.
Are you capturing exception here intentionally? Can't this leave dead app hanging in some failure cases?

File: src/app/mysterion.js [hunk 8]
```diff
@@ -25,26 +27,40 @@ class Mysterion {
...
-      await this.bootstrap()
-      this.buildTray()
+      try {
+        logInfo('Application launch')
+        await this.bootstrap()
+        this.buildTray()
+      } catch (e) {
+        logException('Application launch failed', e)
+        this.bugReporter.main.captureException(e)
+      }
```
Comment (donce) [inline src/app/mysterion.js]: I'm not 100% sure, but I believe that failed hook will terminate application if this hook throws an exception.
Are you capturing exception here intentionally? Can't this leave dead app hanging in some failure cases?

File: src/app/mysterion.js [hunk 9]
```diff
@@ -59,60 +75,71 @@ class Mysterion {
...
     if (this.installer.needsInstallation()) {
+      logInfo("Installing 'mysterium_client' process")
       try {
         await this.installer.install()
       } catch (e) {
-        this.bugReporter.captureException(e)
-        console.error(e)
-        return this.communication.sendRendererShowErrorMessage(translations.daemonInstallationError)
+        this.communication.sendRendererShowErrorMessage(translations.daemonInstallationError)
+        throw new Error("Failed to install 'mysterium_client' process. " + e)
```
Comment (donce) [inline src/app/mysterion.js]: We should not create a new instance of `Error` here - stacktrace of original error is lost here, which provides details of where error happened.

File: src/app/mysterion.js [hunk 10]
```diff
@@ -153,62 +180,64 @@ class Mysterion {
...
-  /**
-   * notifies the renderer that we're good to go and sets up the system tray
-   */
-  startApp () {
+  updateProposalsLoop () {
     this.proposalFetcher.subscribe((proposals) => this.communication.sendProposals(proposals))
-    this.proposalFetcher.start()
+
+    this.monitoring.subscribeUp(() => {
+      logInfo("Subscribing 'mysterium_client' proposals")
```
Comment (donce) [inline src/app/mysterion.js]: This is confusing - nothing is being subscribed here. `Starting proposal fetcher`?

File: src/app/mysterion.js [hunk 11]
```diff
@@ -153,62 +180,64 @@ class Mysterion {
...
+    await this.process.start()
     this.process.onLog(processLogLevels.LOG, (data) => cacheLogs(processLogLevels.LOG, data))
     this.process.onLog(processLogLevels.ERROR, (data) => cacheLogs(processLogLevels.ERROR, data))
+  }
 
-    this.monitoring.start()
-    this.monitoring.onProcessReady(() => {
-      updateRendererWithHealth()
-      this.startApp()
+  startProcessMonitoringLoop () {
```
Comment (donce) [inline src/app/mysterion.js]: why monitoring *loop*? There doesn't seem to be any loop mentioned inside - it's just an implementation detail which is hidden from us. Maybe simply `startProcessMonitoring`?

File: src/libraries/mysterium-client/launch-daemon/process.js [hunk 12]
```diff
@@ -17,9 +17,13 @@ class Process {
   }
 
   start () {
-    this.tequilapi.healthCheck().catch(() => {
-      console.log('touched the daemon, now it should be up')
-    })
+    this.tequilapi.healthCheck()
+      .then(() => {
+        console.log('Touched the daemon, now it must be up')
```
Comment (donce) [inline src/libraries/mysterium-client/launch-daemon/process.js]: `should` sounded better - it's possible that it's not up.

File: src/libraries/mysterium-client/monitoring.js [hunk 13]
```diff
@@ -1,55 +1,95 @@
...
-    if (this.ipcTimeout) {
-      clearTimeout(this.ipcTimeout)
+    if (this._timer) {
+      clearTimeout(this._timer)
     }
   }
 
-  isRunning () {
-    return this.clientIsRunning
+  subscribeStatus (callback: StatusCallback) {
```
Comment (donce) [inline src/libraries/mysterium-client/monitoring.js]: This doesn't seem to be used.

File: src/libraries/mysterium-client/monitoring.js [hunk 14]
```diff
@@ -1,55 +1,95 @@
...
     try {
       await this.api.healthCheck(healthCheckTimeout)
-      this.clientIsRunning = true
+      isRunning = true
     } catch (e) {
-      this.clientIsRunning = false
+      isRunning = false
+    }
+
+    this._notifySubscribers(isRunning)
```
Comment (donce) [inline src/libraries/mysterium-client/monitoring.js]: If any of subscribers throws an error, healthcheck loop will stop.
We keep our healthcheck running even if one of subscribers callbacks failed:
```js
let err = null
try {
  this._notifySubscribers(isRunning)
} catch (e) {
  err = e
}
this._timer = setTimeout(() => this._healthCheckLoop(), healthCheckInterval)
if (err) {
  throw error
}
```

File: src/main/dependencies/modules/mysterium-client.js [hunk 15]
```diff
@@ -34,7 +34,9 @@ function bootstrap (container: Container) {
   container.service(
     'mysteriumClientMonitoring',
     ['tequilapiClient'],
-    (tequilapiClient) => new Monitoring(tequilapiClient)
+    (tequilapiClient) => {
+      return new Monitoring(tequilapiClient)
```
Comment (donce) [inline src/main/dependencies/modules/mysterium-client.js]: Why long form?

File: test/integration/libraries/mysterium-client/standalone/process.spec.js [hunk 16]
```diff
@@ -36,22 +37,44 @@ xdescribe('Standalone Process', () => {
...
+    it('sends log data to callback on()', () => {
+      expect(logs.pop()).to.include('Api started on: ' + tequilapiPort)
+    })
   })
 
-  it('responds to healthcheck with uptime', async () => {
-    const res = await tequilapi.healthCheck()
-    expect(res.uptime).to.be.ok
-    expect(res.version).to.be.ok
+  describe('health checking', () => {
```
Comment (donce) [inline test/integration/libraries/mysterium-client/standalone/process.spec.js]: Are you aware that all these tests are disabled, because of `xdescribe` in this file? Have you tried running them when they are enabled?

File: src/app/mysterion.js (hunk 6 above)
Comment (Waldz) [reply src/app/mysterion.js]: Fixed

File: src/app/mysterion.js (hunk 7 above)
Comment (Waldz) [reply src/app/mysterion.js]: Code is tested.
Exception is captured intentionally, so Sentry knows about all fatal failures

File: src/app/mysterion.js (hunk 8 above)
Comment (Waldz) [reply src/app/mysterion.js]: same here

File: src/app/mysterion.js (hunk 9 above)
Comment (Waldz) [reply src/app/mysterion.js]: There is TODO for that, not in PR scope

File: src/app/mysterion.js (hunk 10 above)
Comment (Waldz) [reply src/app/mysterion.js]: Improved

File: src/app/mysterion.js (hunk 11 above)
Comment (Waldz) [reply src/app/mysterion.js]: renamed

File: src/app/mysterion.js (hunk 7 above)
Comment (donce) [reply src/app/mysterion.js]: > Code is tested.
You mean you've tested it manually?

> Exception is captured intentionally, so Sentry knows about all fatal failures
Yes, but it's not re-thrown anymore - that's what is concerning me.

File: src/libraries/mysterium-client/monitoring.js (hunk 14 above)
Comment (Waldz) [reply src/libraries/mysterium-client/monitoring.js]: Fixed

File: src/main/dependencies/modules/mysterium-client.js (hunk 15 above)
Comment (Waldz) [reply src/main/dependencies/modules/mysterium-client.js]: Fixed

File: test/integration/libraries/mysterium-client/standalone/process.spec.js (hunk 16 above)
Comment (Waldz) [reply test/integration/libraries/mysterium-client/standalone/process.spec.js]: Off course

File: src/libraries/mysterium-client/launch-daemon/process.js (hunk 12 above)
Comment (Waldz) [reply src/libraries/mysterium-client/launch-daemon/process.js]: Fixed

File: src/app/mysterion.js (hunk 7 above)
Comment (Waldz) [reply src/app/mysterion.js]: PR review is meant for code review, not assumptions.

Comment (donce) [reply src/app/mysterion.js]: You are changing this behaviour in you PR with code changes, that's why I want to understand is it intentional or not and how does it effect start-up.

File: src/app/mysterion.js [hunk 17]
```diff
@@ -53,18 +68,28 @@ class Mysterion {
...
+      await this._acceptTermsOrQuit()
     }
 
     await this._ensureDaemonInstallation()
-    await this._startProcessAndMonitoring()
+    // if all is good, let's boot up the client
+    this._startProcess()
+    // and start monitoring it
+    this._startProcessMonitoring()
+    // and handle some actions after process is up
```
Comment (donce) [inline src/app/mysterion.js]: These new comments in this block seem redundant - code is pretty self-explanatory.
Perhaps they are result of conflict resolving?

File: src/app/mysterion.js [hunk 18]
```diff
@@ -193,62 +215,62 @@ class Mysterion {
...
 
   /**
-   * notifies the renderer that we're good to go and sets up the system tray
+   * Blocking function until process is launched
    */
-  startApp () {
+  async _onProcessReady (callback) {
+    await onFirstEvent(this.monitoring.subscribeUp.bind(this.monitoring))
+
+    callback()
```
Comment (donce) [inline src/app/mysterion.js]: > This will go to library `mysterium-client`, will be useful for other flavours

How will it be useful? My first though was for backwards-compatibility with callbacks without async/await, but this method is `async`, so we are already required lib users to be able to support async/await.

File: src/libraries/mysterium-client/monitoring.js (hunk 13 above)
Comment (donce) [reply src/libraries/mysterium-client/monitoring.js]: This doesn't seem to be productive - it's against [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).

When are we planning to extract this library?

It seems like useful method, but at the moment, this class is not a library - now it serves for us, and the simpler it is, the faster we'll move.

Once it's a separate library, then we can add additional features, which might be required by other possible users.

File: test/integration/libraries/mysterium-client/standalone/process.spec.js [hunk 19]
```diff
@@ -36,22 +37,44 @@ xdescribe('Standalone Process', () => {
...
+    before(async () => {
+      processMonitoring = new Monitoring(tequilapi)
+      processMonitoring.start()
+    })
+
+    after(() => {
+      processMonitoring.stop()
+    })
+
+    it('monitoring send UP status', (done) => {
```
Comment (donce) [inline test/integration/libraries/mysterium-client/standalone/process.spec.js]: *"it monitoring send UP status"* doesn't make sense to me - *"it sends status UP"* ?

File: test/integration/libraries/mysterium-client/standalone/process.spec.js [hunk 20]
```diff
@@ -36,22 +37,44 @@ xdescribe('Standalone Process', () => {
...
+      processMonitoring.subscribeUp(() => {
+        done()
+      })
```
Comment (donce) [inline test/integration/libraries/mysterium-client/standalone/process.spec.js]: We can pass `done` correctly, no need to create one more function abstraction:
```js
processMonitoring.subscribeUp(done)
```

File: src/app/mysterion.js (hunk 9 above)
Comment (donce) [reply src/app/mysterion.js]: Which TODO are you referring? I'm not talking about some unrelated existing issues - this change brakes stacktrace.

Comment (Waldz) [reply src/app/mysterion.js]: It's old convention in this file with TODO registered. Fix fill come in upcomming separate PR

File: src/libraries/mysterium-client/monitoring.js (hunk 13 above)
Comment (Waldz) [reply src/libraries/mysterium-client/monitoring.js]: it's  enough to understand, because code is laying in `src/libraries/..`. Library migration will come in separate PR

File: src/app/mysterion.js (hunk 17 above)
Comment (Waldz) [reply src/app/mysterion.js]: Thanks. Removed

File: src/app/mysterion.js (hunk 18 above)
Comment (Waldz) [reply src/app/mysterion.js]: Quote not related to code lines

File: test/integration/libraries/mysterium-client/standalone/process.spec.js (hunk 19 above)
Comment (Waldz) [reply test/integration/libraries/mysterium-client/standalone/process.spec.js]: Improved

File: test/integration/libraries/mysterium-client/standalone/process.spec.js (hunk 20 above)
Comment (Waldz) [reply test/integration/libraries/mysterium-client/standalone/process.spec.js]: Improved

**Comment by Waldz** [comment]
I must remind that this PR implements more detailed Mysterion startup logs ONLY

File: src/app/mysterion.js (hunk 18 above)
Comment (Waldz) [reply src/app/mysterion.js]: Did not get about lib users

Comment (Waldz) [reply src/app/mysterion.js]: I just want to collect actions performed after process is ready to single method.
BTW code was the same before PR 

```

```

Comment (donce) [reply src/app/mysterion.js]: I do not get what are you telling... Lets start from beginning - why do you need `async` method with a `callback` param?

> BTW code was the same before PR

Can you direct me to it in `master`? I don't see any `async` methods with callback parameter in removed code.

File: src/app/mysterion.js [hunk 21]
```diff
@@ -193,62 +212,62 @@ class Mysterion {
...
+      this.communication.sendMysteriumClientDown()
     })
+
+    logInfo("Starting 'mysterium_client' monitoring")
+    this.monitoring.start()
   }
 
   /**
-   * notifies the renderer that we're good to go and sets up the system tray
+   * Blocking function until process is launched
```
Comment (shroomist) [inline src/app/mysterion.js]: this comment here is confusing, 
`async _onProcessReady` is non-blocking,
I think this method signature is self-explanatory.

File: src/app/mysterion.js [hunk 22]
```diff
@@ -193,62 +212,62 @@ class Mysterion {
...
    */
-  startApp () {
+  async _onProcessReady (callback) {
+    await onFirstEvent(this.monitoring.subscribeUp.bind(this.monitoring))
```
Comment (shroomist) [inline src/app/mysterion.js]: to simplify this, you could do
```
_onProcessReady (callback) {
   onFirstEvent(this.monitoring.subscribeUp.bind(this.monitoring)).then(callback)
```

File: src/libraries/mysterium-client/monitoring.js (hunk 13 above)
Comment (shroomist) [inline src/libraries/mysterium-client/monitoring.js]: I suggest `subscribeStatus` => `onStatusChange(callback)` or `subscribeStatusChange`

File: src/app/mysterion.js (hunk 22 above)
Comment (donce) [reply src/app/mysterion.js]: Exactly.

File: src/app/mysterion.js (hunk 18 above)
Comment (ignasbernotas) [reply src/app/mysterion.js]: @Waldz _onProcessReady is only used in that single place, and `this.communication.sendMysteriumClientIsReady()` can be moved to the method `_onProcessReady` itself, no need for the callback since it's not being reused:)

What you're referring to in master is a different case, `this.monitoring.onProcessReady` was called directly, and since this method is being called multiple times in different places it makes sense for it to have a callback function as an argument (since many different places need to execute some logic during this event), but it isn't necessary to pass a callback in a small abstraction method (`_onProcessReady`) that isn't reused.

File: src/libraries/mysterium-client/monitoring.js (hunk 13 above)
Comment (ignasbernotas) [reply src/libraries/mysterium-client/monitoring.js]: 👍 for `onStatusChange`

File: src/app/mysterion.js [hunk 23]
```diff
@@ -150,27 +171,25 @@ class Mysterion {
...
   // declining terms will quit the app
   async _acceptTermsOrQuit () {
+    logInfo('Accepting terms')
     try {
       const accepted = await this._acceptTerms()
       if (!accepted) {
-        console.log('Terms were refused. Quitting.')
+        logInfo('Terms were refused. Quitting.')
         app.quit()
-        return false
```
Comment (donce) [inline src/app/mysterion.js]: We should terminate `bootstrap` instantly if user declines terms, we shouldn't start-up mysterium client process and other things. AFAIK, `app.quit()` does not stop bootstrap.

File: src/app/mysterion.js [hunk 24]
```diff
@@ -193,62 +212,62 @@ class Mysterion {
...
-    }
+  async _startProcess () {
     const cacheLogs = (level, data) => {
       this.communication.sendMysteriumClientLog({level, data})
       this.bugReporter.pushToLogCache(level, data)
     }
 
-    this.process.start()
+    logInfo("Starting 'mysterium_client' process")
+    await this.process.start()
```
Comment (donce) [inline src/app/mysterion.js]: `start` is not async, you don't have to `await` it.

File: src/app/mysterion.js (hunk 22 above)
Comment (donce) [inline src/app/mysterion.js]: Still not resolved.

File: src/libraries/mysterium-client/monitoring.js [hunk 25]
```diff
@@ -1,55 +1,95 @@
...
+      this._notifySubscribersUp()
+    } else {
+      this._notifySubscribersDown()
     }
+    this._lastIsRunning = isRunning
+  }
+
+  _notifySubscribersStatus (isRunning: boolean) {
+    this._subscribersStatus.forEach((callback: StatusCallback) => {
+      try { callback(isRunning) } catch (e) {}
```
Comment (donce) [inline src/libraries/mysterium-client/monitoring.js]: We shouldn't ignore errors of listeners - we should `logger.info` and `bugReporter.captureInfoException` them.
Same applies for `_notifySubscribersUp` and `_notifySubscribersDown`

File: src/app/mysterion.js (hunk 24 above)
Comment (Waldz) [reply src/app/mysterion.js]: Fixed

File: src/app/mysterion.js (hunk 21 above)
Comment (Waldz) [reply src/app/mysterion.js]: Fixed

File: src/app/mysterion.js (hunk 22 above)
Comment (Waldz) [reply src/app/mysterion.js]: Very exact feedback, fixed

File: src/app/mysterion.js (hunk 18 above)
Comment (Waldz) [reply src/app/mysterion.js]: I need callback, because my intention is to build self explainable code, which makes clear difference between:
- what is being loaded parallely
- what is being loader after process ready

File: src/libraries/mysterium-client/monitoring.js (hunk 25 above)
Comment (Waldz) [reply src/libraries/mysterium-client/monitoring.js]: Not my scope. Fixing in #184

File: src/libraries/mysterium-client/monitoring.js (hunk 13 above)
Comment (Waldz) [reply src/libraries/mysterium-client/monitoring.js]: Fixing in #184

File: src/libraries/mysterium-client/launch-daemon/process.js [hunk 26]
```diff
@@ -18,7 +18,6 @@
 import {Tail} from 'tail'
 import path from 'path'
 import logLevels from '../log-levels'
-import logger from '../../../app/logger'
```
Comment (ignasbernotas) [inline src/libraries/mysterium-client/launch-daemon/process.js]: why was this removed?

File: src/renderer/App.vue [hunk 27]
```diff
@@ -92,32 +92,26 @@
...
+      this.rendererCommunication.onMysteriumClientDown(() => {
+        this.$store.dispatch('setClientRunningState', false)
 
-        this.$store.dispatch(type.OVERLAY_ERROR, {
-          message: 'mysterium_client is down',
-          hint: 'Please give it a moment to boot. If this message persists try restarting the app or please contact support'
-        })
-        this.$store.dispatch('setClientRunningState', clientRunningState)
+        // do nothing while on terms page
+        if (this.$route.name !== 'terms') {
```
Comment (ignasbernotas) [inline src/renderer/App.vue]: I would prefer:
`if (this.$route.name == 'terms') { return; }`

File: src/libraries/mysterium-client/launch-daemon/process.js (hunk 26 above)
Comment (Waldz) [reply src/libraries/mysterium-client/launch-daemon/process.js]: Because it's `mysterium-client` library and library should not depend on Mysterion application

File: src/renderer/App.vue (hunk 27 above)
Comment (Waldz) [reply src/renderer/App.vue]: Too small improvement to worry about. Such conditional behaviour should be dropped at all

File: src/app/mysterion.js [hunk 28]
```diff
@@ -232,65 +253,58 @@ class Mysterion {
     } catch (e) {
       const error = new Error(translations.termsAcceptError)
       error.original = e
-      logger.error(error)
       throw error
     }
-
-    this.window.resize(this.config.windows.app)
```
[the remark this answers is not in the document]
Comment (Waldz) [reply src/app/mysterion.js]: Moved to upper layer

File: src/app/mysterion.js [hunk 29]
```diff
@@ -189,27 +212,25 @@ class Mysterion {
...
   // declining terms will quit the app
   async _acceptTermsOrQuit () {
+    logInfo('Accepting terms')
     try {
       const accepted = await this._acceptTerms()
       if (!accepted) {
-        logger.info('Terms were refused. Quitting.')
+        logInfo('Terms were refused. Quitting.')
         app.quit()
-        return false
```
[the remark this answers is not in the document]
Comment (Waldz) [reply src/app/mysterion.js]: Restored previous code

File: src/libraries/mysterium-client/launch-daemon/process.js [hunk 30]
```diff
@@ -46,7 +49,7 @@ class Process {
 
   async stop () {
     await this.tequilapi.stop()
-    logger.info('Client Quit was successful')
+    console.log('Client Quit was successful')
```
[the remark this answers is not in the document]
Comment (Waldz) [reply src/libraries/mysterium-client/launch-daemon/process.js]: Refer to https://github.com/MysteriumNetwork/mysterion/pull/161#discussion_r192098875

File: src/app/mysterion.js [hunk 31]
```diff
@@ -232,65 +259,58 @@ class Mysterion {
...
-      try {
-        this.communication.sendHealthCheck({ isRunning: this.monitoring.isRunning() })
-      } catch (e) {
-        this.bugReporter.captureException(e)
-        return
-      }
-
-      setTimeout(() => updateRendererWithHealth(), 1500)
-    }
+  async _startProcess () {
```
Comment (donce) [inline src/app/mysterion.js]: `async` is not needed anymore.

Comment (Waldz) [reply src/app/mysterion.js]: Removed
